### PR TITLE
fixed relative media URLs for sub/index.md files (fixes #494)

### LINKED ISSUE #494: media in sub/index.md gets invalid relative path
When creating a setup where the layout is:
```
sub/index.md
sub/image.png
```
and sub/index.md content is
```
![](image.png)
```
.. Then media url created is `../image.png`. 

Submitting a pull request to fix this.

### PR DESCRIPTION
**Pull request by viktorasm**
Adding a fix for the issue I raised (see #494). Added some __repr__() methods to nav.py, hope you don't mind.

### CODE REVIEW

**Comment by d0ugal** [comment]
Interesting catch! I hadn't considered an index.md file before - I'm actually a little surprised you don't end up with /index/index.html.

Will review properly in the next few days when I'm back at the computer.

**Comment by viktorasm** [comment]
the reason I needed `article/index.md` is that I had a lot of pictures per article, and I want them to group together with the mardown file, so the structure would be:
```
article1
   index.md
   image1.png
   image2.png
article2
   index.md
   image1.png
   image2.png
```

Also, was having a second thought about how relative URLs are being handled in MkDocs, feels a bit hacky, won't you say? Should really operate with absolute URLs, and only convert to relative as a last step, will remove any ambiguity having a absolute HTML and absolute media URL.

**Comment by d0ugal** [comment]
@viktorasm I'd be happy to see any solid proposals for improving the URL handling.

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,7 @@ htmlcov/
 mkdocs.egg-info/
 *.pyc
 .coverage
+venv
+.pydevproject
+.project
+.settings
```
Comment (d0ugal) [inline .gitignore]: This can go in your global gitignore.

**Comment by d0ugal** [comment]
This is conflicting with master as it came along during a refactor of the navigation. I've tried to rebase it and resolve the conflicts but can't figure it out.

**Pull request closed without merging**